Assets/ComethSDK/Scripts/Tools/SaveLoadPersistentData.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using ComethSDK.Scripts.Types;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace ComethSDK.Scripts.Tools
+{
+	public class SaveLoadPersistentData
+	{
+		/// <summary>
+		///     Save data to a file (overwrite completely)
+		/// </summary>
+		public static void Save(EncryptionData data, string folder, string file)
+		{
+			// get the data path of this save data
+			var dataPath = GetFilePath(folder, file);
+
+			var jsonData = JsonConvert.SerializeObject(data);
+			var byteData = Encoding.ASCII.GetBytes(jsonData);
+
+			// create the file in the path if it doesn't exist
+			// if the file path or name does not exist, return the default SO
+			if (!Directory.Exists(Path.GetDirectoryName(dataPath)))
+				Directory.CreateDirectory(Path.GetDirectoryName(dataPath));
+
+			// attempt to save here data
+			try
+			{
+				// save datahere
+				File.WriteAllBytes(dataPath, byteData);
+				Debug.Log("Save data to: " + dataPath);
+			}
+			catch (Exception e)
+			{
+				// write out error here
+				Debug.LogError("Failed to save data to: " + dataPath);
+				Debug.LogError("Error " + e.Message);
+			}
+		}
+
+		/// <summary>
+		///     Load all data at a specified file and folder location
+		/// </summary>
+		public static EncryptionData Load(string folder, string file)
+		{
+			// get the data path of this save data
+			var dataPath = GetFilePath(folder, file);
+
+			// if the file path or name does not exist, return the default SO
+			if (!Directory.Exists(Path.GetDirectoryName(dataPath)))
+			{
+				Debug.LogWarning("File or path does not exist! " + dataPath);
+				return null;
+			}
+
+			// load in the save data as byte array
+			byte[] jsonDataAsBytes = null;
+
+			try
+			{
+				jsonDataAsBytes = File.ReadAllBytes(dataPath);
+				Debug.Log("<color=green>Loaded all data from: </color>" + dataPath);
+			}
+			catch (Exception e)
+			{
+				Debug.LogWarning("Failed to load data from: " + dataPath);
+				Debug.LogWarning("Error: " + e.Message);
+				return null;
```

Comment:
Do we want to return `null` when it fails? Rapidly looking at callers, it seems like `null` values are used to signal that something went wrong. Can't we just propagate the exception then?
I'm also concerned that it would make debugging less obvious for integrating users. One might just see the last `New Domain detected` message in their logs, caught by the null-check in `EoaFallbackService.GetSigner`, inducing the developer into looking at domain-related issues while it's actually because of the filesystem. Could I get more insights on that matter?

Reply:
Yes absolutely, will propagate the exception to avoid getting the wrong exception !